internal/scope/client.go

```diff
@@ -0,0 +1,30 @@
+package scope
+
+import (
+	"fmt"
+	"net/http"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// RoundTripper satisfies the http.RoundTripper interface and is used to
+// customize the default http client RoundTripper
+type RoundTripper struct {
+	// Default http.RoundTripper
+	Rt http.RoundTripper
+}
+
+// RoundTrip performs a round-trip HTTP request, injecting the OpenStack
+// Request ID header when appropriate
+func (rt *RoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	reconcileID := controller.ReconcileIDFromContext(request.Context())
+	if reconcileID != "" {
+		request.Header.Set("X-OpenStack-Request-ID", "req-"+string(reconcileID))
+	}
+
+	ort := rt.Rt
+	if ort == nil {
+		return nil, fmt.Errorf("rt RoundTripper is nil, aborting")
```

Comment:
I think I would be fine with a panic here. Why would the roundtripper be nil?

Reply:
Right, I've removed the check.